src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,594 @@
+    }
+
+    /**
+     * Creates HTML div element as container for plot.
+     */
+    createDivElem(width,height) {
+        const divElem = document.createElementNS(null,"div");
+        divElem.setAttributeNS(null,"class","vis-scatterplot");
+        divElem.setAttributeNS(null,"viewBox",0 + " " + 0 + " " + width + " " + height);
+        divElem.setAttributeNS(null,"width","100%");
+        divElem.setAttributeNS(null,"height","100%");
+        divElem.setAttributeNS(null,"transform","matrix(1 0 0 -1 0 0)");
+
+        function addStyleToElem(attr,stl) {
+            let style       = document.createElement("style");
+            style.innerText = attr + "{" + stl + "}"
+
+            divElem.appendChild(style);
+        }
+
+        let darkStrokeColor   = `rgba(255,255,255,0.7)`;
+        let buttonLightColor  = `#333`;
+        let darkBtnHoverColor = `rgba(255,255,255,0.5)`;
+        let darkSelectionFill = `#efefef`;
+
+        addStyleToElem('.selection','rx: 4px;stroke: transparent;')
+        addStyleToElem('button',`
+            margin-left: 5px; 
+            margin-bottom: 5px;
+            display: inline-block;
+            padding: 2px 10px;
+            outline: none;
+            background-color: transparent;
+            border: 1px solid ${buttonLightColor};
+            color: ${buttonLightColor};
+            border-radius: 14px;
+            font-size: 10px;
+            font-family: DejaVuSansMonoBook;
+            vertical-align: top;
+            transition: all 0.3s ease;
+        `)
+        addStyleToElem('button:hover',`
+            background-color: ${buttonLightColor};
+            color: ${darkSelectionFill};
+        `)
+
+        addStyleToElem('.dark button',`
+            border: 0;
+            background-color: ${darkStrokeColor};
+        `)
+        addStyleToElem('.dark button:hover',`
+            background-color: ${darkBtnHoverColor};
+        `)
+        addStyleToElem('.dark .selection',`fill: ${darkSelectionFill}`)
+        addStyleToElem('.dark line',`stroke: ${darkStrokeColor};`)
+        addStyleToElem('.dark .domain',`stroke: ${darkStrokeColor};`)
+        addStyleToElem('.dark text',`fill: ${darkStrokeColor};`)
 
-        this.dom.appendChild(divElem);
         return divElem;
     }
 
+    /**
+     * Helper function for button creation.
+     */
+    createBtnHelper() {
+        const btn = document.createElement("button");
+        btn.setAttribute("width","80px");
+        btn.setAttribute("height","20px");
+        return btn
+    }
+
+    /**
+     * Creates a button to fit all points on plot.
+     */
+    createButtonFitAll(scaleAndAxis,scatter,points,extremesAndDeltas,zoom,box_width) {
+        const btn = this.createBtnHelper()
+
+        let text = document.createTextNode("Fit all");
+        btn.appendChild(text);
+
+        let self = this;
+        function unzoom() {
+            zoom.zoomElem.transition().duration(0).call(zoom.zoom.transform,d3.zoomIdentity);
```

Comment:
is this diabling unzoom animation? If so, let it be there! We love animations, we hate "view jumps"! :)